### PR TITLE
EXTRA_OPTIONS are appended to exisiting JVM options

### PR DESCRIPTION
**Pull request by annaibm**
Append user specified jvmOpts to exisitng jvm options

resolves : https://github.com/adoptium/aqa-tests/issues/4849

### CODE REVIEW

File: jck/jtrunner/JavaTestRunner.java [hunk 1]
```diff
@@ -892,7 +893,8 @@ public static boolean jckConfigurationForMultijvmWithNoAgent() throws Exception
 			// Get any additional jvm options for specific tests.
 			extraJvmOptions += getTestSpecificJvmOptions(jckVersion, tests);
 
-			extraJvmOptions += suppressOutOfMemoryDumpOptions;
+			extraJvmOptions += suppressOutOfMemoryDumpOptions + " " + jvmOpts;
+			
 
 			// Add the JVM options supplied by the user plus those added in this method to the jtb file option.
 			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " \"" + ";\n";
```
Comment (llxia) [inline jck/jtrunner/JavaTestRunner.java:900]: This PR is not quite right. The EXTRA_OPTIONS is not added at the end of all other options.

I would suggest the following:
- revert the changes in the PR except line 494
- line 172
```
jvmOpts = System.getProperty("other.opts").trim() + " " + System.getProperty("jvm.options");
```
- in line 898
```
fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + jvmOpts + " \"" + ";\n";
```

**Review by llxia — CHANGES_REQUESTED** [review]

**Comment by annaibm** [comment]
Grinder_JCK  output:
#1906

File: jck/jtrunner/JavatestUtil.java [hunk 2]
```diff
@@ -177,7 +177,12 @@ public static void main(String args[]) throws Exception {
 			} 
 		}
 		
-		jvmOpts = System.getProperty("jvm.options").trim() + " " + System.getProperty("other.opts");
+		if (System.getProperty("other.options") != null) {
+			jvmOpts = System.getProperty("other.options").trim() + " " + System.getProperty("jvm.options");
+		}
+		else {
+			jvmOpts = System.getProperty("jvm.options");
+		}
```
Comment (llxia) [inline jck/jtrunner/JavatestUtil.java:185]: This is not quite correct. We need to check `jvm.options` as well.

```
if (System.getProperty("other.options") != null) {
   jvmOpts += System.getProperty("other.options").trim() + " ";
}
if (System.getProperty("jvm.options") != null) {
   jvmOpts += System.getProperty("jvm.options").trim() + " ";
}

```

File: jck/jtrunner/JavatestUtil.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public class JavatestUtil {
 	private static String testExecutionType;
 	private static String withAgent;
 	private static String interactive;
-	private static String extraJvmOptions;
+	private static String extraJvmOptions= "";
```
Comment (llxia) [inline jck/jtrunner/JavatestUtil.java:54]: add a space before `=`

File: jck/jtrunner/JavatestUtil.java [hunk 4]
```diff
@@ -839,9 +845,10 @@ private static boolean generateJTB() throws Exception {
 			// Get any additional jvm options for specific tests.
 			extraJvmOptions += getTestSpecificJvmOptions(jckVersion, tests);
 			extraJvmOptions += suppressOutOfMemoryDumpOptions;
+			
 
 			// Add the JVM options supplied by the user plus those added in this method to the jtb file option.
-			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " \"" + ";\n";	
+			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " " + jvmOpts + " \"" + ";\n";
```
Comment (llxia) [inline jck/jtrunner/JavatestUtil.java:851]: https://github.com/adoptium/aqa-tests/blob/6c6b9a5faaee7d1f8a2a3fb861fde3a1c554533b/jck/jtrunner/JavatestUtil.java#L678 should be updated as well.

File: jck/jtrunner/JavaTestRunner.java [hunk 5]
```diff
@@ -168,8 +168,7 @@ public static void main(String args[]) throws Exception {
 				}
 			} 
 		}
-		
-		jvmOpts = System.getProperty("jvm.options").trim() + " " + System.getProperty("other.opts"); 
+		jvmOpts = System.getProperty("other.options").trim() + " " + System.getProperty("jvm.options");
```
Comment (llxia) [inline jck/jtrunner/JavaTestRunner.java:171]: Same as JavatestUtil. This needs to be updated

File: jck/jtrunner/JavaTestRunner.java [hunk 6]
```diff
@@ -893,9 +893,10 @@ public static boolean jckConfigurationForMultijvmWithNoAgent() throws Exception
 			extraJvmOptions += getTestSpecificJvmOptions(jckVersion, tests);
 
 			extraJvmOptions += suppressOutOfMemoryDumpOptions;
+			
 
 			// Add the JVM options supplied by the user plus those added in this method to the jtb file option.
-			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " \"" + ";\n";	
+			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " " + jvmOpts + " \"" + ";\n";
```
Comment (llxia) [inline jck/jtrunner/JavaTestRunner.java:899]: https://github.com/adoptium/aqa-tests/blob/6c6b9a5faaee7d1f8a2a3fb861fde3a1c554533b/jck/jtrunner/JavaTestRunner.java#L733 should be updated as well.

**Review by llxia — CHANGES_REQUESTED** [review]

**Comment by annaibm** [comment]
Grinder Output:
1907

**Comment by llxia** [comment]
@annaibm Please refer to https://github.com/eclipse-openj9/openj9/blob/master/CONTRIBUTING.md#commit-guidelines. We do not need to put code in the commit message. Also, could you update the description `related: aqa-tests/issues/4849` above? Thanks

**Review by llxia — APPROVED** [review]
Thanks @annaibm

**Comment by smlambert** [comment]
Publicly visible grinder: https://ci.adoptium.net/view/Test_grinder/job/Grinder/8026

**Comment by llxia** [comment]
@smlambert could you review this PR when you have time? Thanks

**Review by smlambert — APPROVED** [review]
thanks @annaibm !

**Comment by annaibm** [comment]
Thank you @llxia and @smlambert!

**Pull request merged**